configure

```diff
@@ -0,0 +1,404 @@
+fi
+if [ -z "${TEST_LDFLAGS+IS_SET}" ]; then
+    TEST_LDFLAGS=""
+    AUTO_TEST_LDFLAGS="true"
+else
+    AUTO_TEST_LDFLAGS="false"
+fi
+
+## Verify PLATFORM value
+if [ "$PLATFORM" != "Linux" ] && [ "$PLATFORM" != "FreeBSD" ] && \
+[ "$PLATFORM" != "OpenBSD" ] && [ "$PLATFORM" != "Darwin" ]; then
+    warning "$PLATFORM" platform is unknown!
+    sub_warning Known Platforms are: Linux, FreeBSD, OpenBSD, Darwin
+fi
+
+## Create testfile.cc to test c++ compiler
+echo "int main(int argc, char **argv) { return 0; }" > testfile.cc || error Cant create test file
+
+## Find and test C++ compiler
+if [ -z "${CXX:-}" ]; then
+    info Checking C++ compiler...
+    for guess in g++ clang++ c++; do
+        findcxx "$guess"
+        if [ -n "${CXX:-}" ]; then
+            sub_info "$CXX"
+            break # Found
+        fi
+    done
+    if [ -z "${CXX:-}" ]; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+cxx_works "$CXX"
+if [ -n "${CXX_FOR_BUILD:-}" ]; then
+    cxx_works "$CXX_FOR_BUILD"
+fi
+
+## Test compiler/linker supported arguments
+if [ "$AUTO_CXXFLAGS" = "true" ]; then
+    for argument in -std=c++11 \
+                    -Wall \
+                    -Os \
+                    -fno-plt
+    do
+        try_cxx_argument CXXFLAGS $argument
+    done
+    if [ "$PLATFORM" != "Darwin" ]; then
+        try_cxx_argument CXXFLAGS -fno-rtti
+    fi
+fi
+if [ "$AUTO_LDFLAGS" = "true" ] && [ "$AUTO_CXXFLAGS" = "true" ]; then
+    try_both_argument CXXFLAGS LDFLAGS -flto
+fi
+if [ "$AUTO_LDFLAGS" = "true" ]; then
+    if [ "$PLATFORM" = "FreeBSD" ]; then
+        try_ld_argument LDFLAGS -lrt
+    fi
+fi
+if [ "${NO_SANITIZE:-}" = "auto" ]; then
+    if [ "$AUTO_TEST_LDFLAGS" = "true" ] && [ "$AUTO_TEST_CXXFLAGS" = "true" ]; then
+        try_both_argument TEST_CXXFLAGS TEST_LDFLAGS -fsanitize=address,undefined
+    elif [ "$AUTO_TEST_LDFLAGS" = "true" ]; then
+        try_ld_argument TEST_LDFLAGS -fsanitize=address,undefined
+    elif [ "$AUTO_TEST_CXXFLAGS" = "true" ]; then
+        try_cxx_argument TEST_CXXFLAGS -fsanitize=address,undefined
+    fi
+elif [ "${NO_SANITIZE:-}" = "0" ]; then
+    TEST_CXXFLAGS="$TEST_CXXFLAGS -fsanitize=address,undefined"
+    TEST_LDFLAGS="$TEST_LDFLAGS -fsanitize=address,undefined"
+fi
+
+## Create mconfig
+rm -f testfile*
+info Creating mconfig...
+cat << EOF > mconfig
+## Auto-generated by "$0" for "$PLATFORM"
+# All changes will be lost if "$0" will be runned.
+# Installation path options.
+
+SBINDIR=$SBINDIR
+MANDIR=$MANDIR
+SYSCONTROLSOCKET=$SYSCONTROLSOCKET
+
+# General build options.
+
+# Linux (GCC): Note with GCC 5.x/6.x you must use the old ABI, with GCC 7.x you must use
+# the new ABI. See BUILD file for more information.
+# MacOS: Cannot use -fno-rtti: apparently prevents exception handling from working properly.
+# FreeBSD: Cannot use LTO with default linker.
+
+CXX=$CXX
+CXXFLAGS=$CXXFLAGS $CXXFLAGS_EXTRA
+TEST_CXXFLAGS=$CXXFLAGS $TEST_CXXFLAGS $TEST_CXXFLAGS_EXTRA
```

Comment:
So the `TEST_CXXFLAGS` used for build will always contain `CXXFLAGS` _as well as_ `TEST_CXXFLAGS` (and `TEST_CXXFLAGS_EXTRA`) passed into configure? Isn't the point of TEST_CXXFLAGS when passed into configure to be able to override the normal flags?

Reply:
So I've thought about this a little. Maybe you shouldn't accept `TEST_CXXFLAGS`/`TEST_LDFLAGS` as user input in the script? You can accept `TEST_CXXFLAGS_EXTRA`/`TEST_LDFLAGS_EXTRA` only, for example. That would make things a lot simpler.

Otherwise you should:

- fix: not include `CXXFLAGS` in the output `TEST_CXXFLAGS`, and
- fix: not include `LDFLAGS` in the output `TEST_LDFLAGS`, and
- ensure that `--enable-sanitize` does something sensible when `TEST_CXXFLAGS` and/or `TEST_LDFLAGS` are specified directly by the user. I.e. adds the flags anyway (with a message printed) or does nothing (with a message printed).

User experience is important! If something looks like it should work, then either it should work, or it should at least give feedback rather than silently not working.

Eg consider right now if I:

```
./configure TEST_CXXFLAGS="" TEST_LDFLAGS="" --enable-sanitizer
```

There's nothing in the help output that suggests this won't work - but it won't!

Reply:
I removed `--enable-sanitize` option and `TEST_CXXFLAGS` doesn't include `CXXFLAGS`. but seems like it's break Unit tests on MacOS.

Reply:
The _default_ TEST_CXXFLAGS should be to use CXXFLAGS. You need something like:

```
if [ "$AUTO_TEST_LDFLAGS" = "true" ]; then
    TEST_LDFLAGS=$LDFLAGS
fi
if [ "$AUTO_TEST_CXXFLAGS" = "true" ]; then
    TEST_CXXFLAGS=$CXXFLAGS
fi
```

... between lines 330-331.